### PR TITLE
Switched client to JSON API from XML API

### PR DESCRIPTION
**Pull request by ekoval**
Closes: #22, #23

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: teamsupport/__init__.py [hunk 1]
```diff
@@ -12,6 +12,9 @@
 __all__ = (Action, TeamSupportService, Ticket, config)
 
 
-def init(org_id, auth_key):
+def init(org_id, auth_key, default_ticket_type='Support',
+        default_ticket_status='New'):
```
Comment (snitch) [inline teamsupport/__init__.py]: E128 continuation line under-indented for visual indent

File: teamsupport/__init__.py [hunk 2]
```diff
@@ -12,6 +12,9 @@
 __all__ = (Action, TeamSupportService, Ticket, config)
 
 
-def init(org_id, auth_key):
+def init(org_id, auth_key, default_ticket_type='Support',
+            default_ticket_status='New'):
```
Comment (snitch) [inline teamsupport/__init__.py]: E127 continuation line over-indented for visual indent

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: README.rst [hunk 3]
```diff
@@ -47,3 +47,12 @@ Usage
 .. |Latest Version| image:: https://img.shields.io/pypi/v/teamsupport.svg?style=flat-square
    :target: https://warehouse.python.org/project/teamsupport
 
+
+Running Tests
+-------------
+
+Set correct data in teasmsupport/config.py
```
Comment (highflyingaction) [inline README.rst]: Typo in `teasmsupport`

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: teamsupport/models.py [hunk 4]
```diff
@@ -7,15 +7,21 @@
 from teamsupport.errors import MissingArgumentError
 
 
-class XmlModel(object):
+def _find_dict_by_key_value(list_of_dicts, **kwargs):
+    for dct in list_of_dicts:
+        for (key, value) in kwargs.items():
+            if dct.get(key) == value:
+                return dct
```
Comment (highflyingaction) [inline teamsupport/models.py]: It looks like this returns the first time a kwarg matches, not only if all kwargs match.  Is that the intended behavior?  If it should be matching all kwargs, this might be a good place to use an `else` clause on the `for (key, value) in kwargs.items():` loop.
It is accepts multiple kwargs maybe it should be renamed to indicate it matches on multiple items?
`_find_dict_with_items`?
`_find_dict_with_key_value_pairs`?
`_find_dict_matching_items`?
`_find_dict_matching_key_value_pairs`?

File: teamsupport/models.py [hunk 5]
```diff
@@ -43,7 +49,9 @@ def __init__(self, ticket_id=None, data=None):
             raise MissingArgumentError(
                 "__init__() needs either a 'ticket_id' or 'data' argument "
                 '(neither given)')
-        self.id = self.TicketID
+
+        id_attr = 'ID' if 'ID' in self.data else 'TicketID'
```
Comment (highflyingaction) [inline teamsupport/models.py:55]: When is this `ID` instead of `TicketID`?

File: teamsupport/models.py [hunk 6]
```diff
@@ -58,15 +66,16 @@ def create(cls, user_email, user_first_name, user_last_name, title,
...
             'TicketStatusID': cls._get_ticket_status_id(
-                config.DEFAULT_TICKET_STATUS),
-            'TicketTypeID': cls._get_ticket_type_id(
-                config.DEFAULT_TICKET_TYPE),
-            'ContactID': contact.id
+                ticket_type_id, config.DEFAULT_TICKET_STATUS),
+            'TicketTypeID': ticket_type_id,
+            'IsVisibleOnPortal': True,
         }
-        data.update(params)
```
Comment (highflyingaction) [inline teamsupport/models.py]: Why was this removed? If it is being removed shouldn't the `**param` parameter also be removed?

File: CHANGELOG.rst [hunk 7]
```diff
@@ -1,6 +1,15 @@
...
 =========
 
+0.5.0 (2018-07-05)
+------------------
+
+- Switched to JSON API.
+- Set correct ticket status when creating a ticket
+- Added ability to modify default ticket type and status when calling
+  teamsupport.init()
+
```
Comment (highflyingaction) [inline CHANGELOG.rst:11]: This version bump might also be a good time to drop support for python 3.3.

File: teamsupport/models.py (hunk 5 above)
Comment (ekoval) [reply teamsupport/models.py:55]: Sometimes. As I understand, when Create result is returned, it is 'ID'. When we search results, this is `<entity_name>ID`. Weird, yes.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: CHANGELOG.rst (hunk 7 above)
Comment (ekoval) [reply CHANGELOG.rst:11]: Done

File: teamsupport/models.py (hunk 4 above)
Comment (ekoval) [reply teamsupport/models.py]: Thanks for the good catch! Fixed!

File: CHANGELOG.rst (hunk 7 above)
Comment (highflyingaction) [reply CHANGELOG.rst:11]: 3.3 is also specified in travis.yml https://github.com/yola/teamsupport-python/blob/f6dfe79e78b85a60ce43fbe5203dcd6f5b10f418/.travis.yml#L10-L11
and setup.py https://github.com/yola/teamsupport-python/blob/f6dfe79e78b85a60ce43fbe5203dcd6f5b10f418/setup.py#L59

Python 3.5 should probably also be added as a classifier in setup.py

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: CHANGELOG.rst (hunk 7 above)
Comment (ekoval) [reply CHANGELOG.rst:11]: Added

**Review by highflyingaction — APPROVED** [review]
